src/main/proto/cosmas.proto

```diff
@@ -92,6 +96,7 @@ message CreatePatchResponse {
 message FileVersionListRequest {
     ProjectInfo info = 1;
     string fileId = 2;
+    int32 count = 3; // if count is 0 or -1 response will contain all versions

```

Comment:
So the bigger is the count, the slower and bigger is the response. I suggest adding offset, so that pagination was more efficient.